### ISSUE TITLE
Pushed params always String

### ISSUE THREAD

**Issue by mozochek** [SchabanBo/qlevar_router#47]
Any params, that pushed with route via ```QR.navigator.pushNamed``` has type String and  method ```valueAs<T>()``` did not cast it to initial type.

**Comment by SchabanBo** [comment]
Hi @mozochek 
could you try this again in V1.5.7
and if did not work could you provide an example?

**Comment by mozochek** [comment]
> Hi @mozochek could you try this again in V1.5.7 and if did not work could you provide an example?

Same behaviour on 1.5.7. 

Example is simple:

```dart
// 1. Define some class
class SomeInfo {
  final int a;
  final String b;
  ...
}

// 2. Push new screen via QR
QR.navigator.pushName('myScreenName', params: {'info': SomeInfo(10, 'asd'),},);

// 3. Try to get provided params
@override
void initState() {
  super.initState();

  QR.params['info']?.value // will be string
  QR.params['info']?.valueAs<SomeInfo?>() // will be exception (String can't cast to SomeInfo)
  QR.params.asValueMap['info']?.value // will be string
  QR.params.asValueMap['info']?.valueAs<SomeInfo?>() // will be exception (String can't cast to SomeInfo)
}
```

I think this problem exist because you use ```Uri``` class and its ```queryParameters``` getter, which return ```Map<String, String>```

**Comment by SchabanBo** [comment]
Hi @mozochek 
Found it.
Could you confirm the fix in V1.5.8?

**Comment by mozochek** [comment]
> Hi @mozochek Found it. I wrote a [test](https://github.com/SchabanBo/qlevar_router/blob/78c1572183dac013a333b2e7002b37c7bc7c6cde/test/param_test.dart#L154) for it too. Could you confirm the fix in V1.5.8?

Now it works fine, thx
